server/game/Card.js

```diff
@@ -516,12 +528,16 @@ class Card extends EffectSource {
         });
     }
 
+    canPlayAsUpgrade() {
+        return this.effects.filter(effect => effect.type === 'canPlayAsUpgrade');
```

Comment:
this should be `this.anyEffect('canPlayAsUpgrade')`

Reply:
Hoo boy this was somehow masking some bugs. Fixing...